### DISCUSSION TITLE
Have some ways to match a c struct use rg?

### DISCUSSION THREAD

**Discussion by Lerr1uqs** [BurntSushi/ripgrep#2410] [Q&A] [answered]
like something :
```c
typedef A{
  int bl;
}B;
```
I tried -U option but still not matched multi-line...

**Comment by BurntSushi** [comment, marked as answer]
When asking a question and say "I tried ..." but don't actually elaborate on _what_ you tried, it makes answering the question much harder.

In any case, this will match the specific input you gave:

```
$ cat test.c
typedef A{
  int bl;
}B;
$ rg -U 'typedef A\{(?s:.)*?\}.*;' test.c
1:typedef A{
2:  int bl;
3:}B;
```

But of course, this is not going to match any `typedef`. Just the one you gave and ones very similar to it.
